### PR TITLE
Feat: add comparison_to_empty

### LINKED ISSUE #74: Add `comparison_to_empty`
# What it does

Checks for comparing to an empty slice such as  [], and suggests using `.is_empty()` where applicable.

# Why is this bad?

Some structures can answer `.is_empty()` much faster than checking for equality. So it is good to get into the habit of using `.is_empty()`, and having it is cheap. Besides, it makes the intent clearer than a manual comparison in some contexts.

# Example

```cairo
if s == ArrayTrait::new() {
    ..
}
```

Use instead:

```cairo
if arr.is_empty() {
    ..
}
```

### PR DESCRIPTION
**Pull request by Benjtalkshow**
closes #74 
@0xLucqs 
This is a draft PR 
## Implementation
Checks for comparing to an empty slice such as [], and suggests using .is_empty() where applicable.

### CODE REVIEW

File: crates/cairo-lint-core/src/lints/comparison_to_empty.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        let func_path = func_call.path(db);
+        func_path.as_syntax_node().get_text_without_trivia(db) == "ArrayTrait::new"
+    } else {
+        false
+    }
+}
+
+fn create_diagnostic(message: &str, stable_ptr: SyntaxStablePtrId, severity: Severity) -> PluginDiagnostic {
+    PluginDiagnostic { stable_ptr, message: message.to_string(), severity }
+}
```
Comment (0xLucqs) [inline crates/cairo-lint-core/src/lints/comparison_to_empty.rs:42]: don't need this function, directly create the diagnostic

File: crates/cairo-lint-core/src/lints/mod.rs [hunk 2]
```diff
@@ -11,3 +11,4 @@ pub mod loops;
 pub mod manual;
 pub mod panic;
 pub mod single_match;
+// pub mod comparison_to_empty;
```
Comment (0xLucqs) [inline crates/cairo-lint-core/src/lints/mod.rs:14]: uncomment that

**Review by 0xLucqs — CHANGES_REQUESTED** [review]

**Comment by Benjtalkshow** [comment]
Alright @0xLucqs

**Comment by Benjtalkshow** [comment]
@0xLucqs  i have made the requested changes

**Comment by 0xLucqs** [comment]
fix clippy and don't forget to format

**Comment by Benjtalkshow** [comment]
> fix clippy and don't forget to format

ok

**Comment by 0xLucqs** [comment]
you need to add the tests to the test file similarly to how it's done with others

**Comment by Benjtalkshow** [comment]
> fix the tests, remove the useless added file and restore the cargo.lock plz

alright

**Pull request closed without merging**